frontend/src/concepts/pipelines/topology/usePipelineTaskTopology.ts

```diff
@@ -2,23 +2,37 @@ import { PipelineRunKFv2, PipelineSpecVariable } from '~/concepts/pipelines/kfTy
 import { createNode } from '~/concepts/topology';
 import { PipelineNodeModelExpanded } from '~/concepts/topology/types';
 import { createArtifactNode } from '~/concepts/topology/utils';
+import { usePipelineRunMlmdContext } from '~/concepts/pipelines/apiHooks/mlmd/usePipelineRunMlmdContext';
+import { useExecutionsFromContext } from '~/concepts/pipelines/apiHooks/mlmd/useExecutionsFromContext';
+import { isPipelineRunComplete } from '~/concepts/pipelines/apiHooks/usePipelineRunById';
 import {
   composeArtifactType,
   parseComponentsForArtifactRelationship,
   parseInputOutput,
-  parseRuntimeInfo,
+  parseRuntimeInfoFromExecutions,
+  parseRuntimeInfoFromRunDetails,
   parseTasksForArtifactRelationship,
   parseVolumeMounts,
   translateStatusForNode,
 } from './parseUtils';
 import { KubeFlowTaskTopology } from './pipelineTaskTypes';
 
-const EMPTY_STATE: KubeFlowTaskTopology = { taskMap: {}, nodes: [] };
+const EMPTY_STATE: KubeFlowTaskTopology = { taskMap: {}, nodes: [], loaded: true };
```

Comment:
Are we sure we want an empty state to be considered "loaded"? Would there be a request made and finished at the point of this object being returned?